covalent/cloud_resource_manager/core.py

```diff
@@ -389,7 +397,24 @@ def _get_tf_statefile_path(self) -> str:
 
         """
         # Saving in a directory which doesn't get deleted on purge
-        return str(Path(self.executor_tf_path) / "terraform.tfstate")
+        return str(Path(self.executor_tfstate_path) / "terraform.tfstate")
```

Comment:
>Saving in a directory which doesn't get deleted on purge

`self.executor_tfstate_path` seems to point to `~/.config/covalent/executor_plugins/*`, this WILL get deleted by purge. We need to add an exception in the purge command to prevent deletion of these state files.